lib/ansible/modules/monitoring/icinga2_host.py

```diff
@@ -0,0 +1,332 @@
+        if info['status'] >= 400:
+            body = info['body']
+        return {'code': info['status'], 'data': body}
+
+    def check_connection(self):
+        ret = self.call_url('v1/status')
+        if ret['code'] == 200:
+            return True
+        return False
+
+    def exists(self, hostname):
+        data = {
+            "filter": "match(\"" + hostname + "\", host.name)",
+        }
+        ret = self.call_url(
+            path="v1/objects/hosts",
+            data=self.module.jsonify(data)
+        )
+        if ret['code'] == 200:
+            if len(ret['data']['results']) == 1:
+                return True
+        return False
+
+    def create(self, hostname, data):
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            data=self.module.jsonify(data),
+            method="PUT"
+        )
+        return ret
+
+    def delete(self, hostname):
+        data = {"cascade": 1}
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            data=self.module.jsonify(data),
+            method="DELETE"
+        )
+        return ret
+
+    def modify(self, hostname, data):
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            data=self.module.jsonify(data),
+            method="POST"
+        )
+        return ret
+
+    def diff(self, hostname, data):
+        ret = self.call_url(
+            path="v1/objects/hosts/" + hostname,
+            method="GET"
+        )
+        changed = False
+        ic_data = ret['data']['results'][0]
+        for key in data['attrs']:
+            if key not in ic_data['attrs'].keys():
+                changed = True
+            elif data['attrs'][key] != ic_data['attrs'][key]:
+                changed = True
+        return changed
+
+
+# ===========================================
+# Module execution.
+#
+def main():
+    # use the predefined argument spec for url
+    argument_spec = url_argument_spec()
+    # add our own arguments
+    argument_spec.update(
+        state=dict(default="present", choices=["absent", "present"]),
+        name=dict(required=True, aliases=['host']),
+        zone=dict(),
+        template=dict(default=None),
+        check_command=dict(default="hostalive"),
+        display_name=dict(default=None),
+        ip=dict(required=True),
+        variables=dict(type='dict', default=None),
+    )
+
+    # Define the main module
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    state = module.params["state"]
+    name = module.params["name"]
+    zone = module.params["zone"]
+    template = []
+    template.append(name)
+    if module.params["template"]:
+        template.append(module.params["template"])
+    check_command = module.params["check_command"]
+    ip = module.params["ip"]
+    display_name = module.params["display_name"]
+    if not display_name:
```

Comment:
minor: if a user would set display_name: "", this would not be triggered here. Explicitly checking for none? 

~~~diff
-if not display_name:
+if display_name is not None: 
~~~

Reply:
Accoording too PEP8 (https://www.python.org/dev/peps/pep-0008/#programming-recommendations) checking for empty strings should be done as is writen ; 

For sequences, (strings, lists, tuples), use the fact that empty sequences are false